src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageLoader.cs

```diff
@@ -20,21 +23,210 @@ public DefaultPageLoader(RazorCompiler compiler)
 
         public CompiledPageActionDescriptor Load(PageActionDescriptor actionDescriptor)
```

Comment:
The design of this is greatly improved. All the reflection is here and data flows like `reflection -> descriptor -> thing doer`. The system isn't so coupled.